Library/Homebrew/utils/analytics.rb

```diff
@@ -69,12 +77,67 @@ def report(type, metadata = {})
         nil
       end
 
-      def report_event(category, action, label = os_arch_prefix_ci, value = nil)
-        report(:event,
-               ec: category,
-               ea: action,
-               el: label,
-               ev: value)
+      def report_influx(category, metadata = {})
+        return unless ENV["HOMEBREW_ANALYTICS_ENABLE_INFLUX"]
+
+        # Rewrite to better visualize data
+        metadata[:info]["on_request"] = true if metadata[:on_request] == true
+
+        # Move info one level up
+        tags = []
+        info = metadata.delete(:info)
+        metadata.merge(info).each do |key, value|
+          next unless key
+          next unless value
+
+          tags << "#{key}=#{value.to_s.sub(" ", "\\ ")}"
```

Comment:
So we are just passing through every key/value pair as-is here? I wonder if we want to make this a bit more structured with the callers so querying is easier? 

Similarly, are we concerned about the cardinality of formula values etc.? Are there any InfluxDB guidelines on how much is too much cardinality?

Reply:
Well, we're escaping whitespace. 

> I wonder if we want to make this a bit more structured with the callers so querying is easier?

I'm not sure what you mean by this

> Similarly, are we concerned about the cardinality of formula values etc.? Are there any InfluxDB guidelines on how much is too much cardinality?

I'm not concerned about this right now. I'm not aware of such guidelines, but there won't be that much cardinality with the only differences for tags being in the package name (6000 or so), arch (2), CI (2), developer (2), OS (10 or so), version (however many Homebrew versions release in a year), on_request (2).

Reply:
> > I wonder if we want to make this a bit more structured with the callers so querying is easier?
> 
> I'm not sure what you mean by this

I guess I'm suggesting that this method take specific keyword arguments so that we can ensure they are kept consistent between calls.

It's pretty difficult from the methods (which I wrote originally, hence self-criticism!) which keys/values/tags we're using there.

> but there won't be that much cardinality with the only differences for tags being in the package name (6000 or so)

Yeh, I know very little but from e.g. DataDog usage at GitHub that this would be borderline too high.

> OS (10 or so)

Similarly, may want to restrict this a bit more so it's e.g. just the major/minor version and we omit e.g. random Linux distros.